### ISSUE TITLE
SOInterface Visibility Issue

### ISSUE THREAD

**Issue by commonsguy** [doppllib/SOAndroid#1]
If you follow [the 0.10.5 edition of the `AddingDoppl` instructions](https://github.com/doppllib/doppllib.github.io/blob/0.10.5-update/docs/AddingDoppl.md), and you try to run the `iosApp` when you are done, you get:

```
2017-12-02 09:30:30.228791-0500 iosApp[23354:1020697] *** Terminating app due to uncaught exception
'JavaLangIllegalArgumentException', reason: 'java.lang.IllegalArgumentException:
interface co.doppl.so.api.SOInterface is not visible from class loader'
*** First throw call stack:
(
  0   CoreFoundation                      0x00000001116681ab __exceptionPreprocess + 171
  1   libobjc.A.dylib                     0x000000010e2bef41 objc_exception_throw + 48
  2   doppllib                            0x000000010b29d7f6 JavaLangReflectProxy_getProxyClass0WithJavaLangClassLoader_withIOSClassArray_ + 1791
  3   doppllib                            0x000000010b29d983 JavaLangReflectProxy_newProxyInstanceWithJavaLangClassLoader_withIOSClassArray_withJavaLangReflectInvocationHandler_ + 45
  4   doppllib                            0x000000010b1fb380 -[Retrofit2Retrofit createWithIOSClass:] + 240
  5   doppllib                            0x000000010b234115 SOARepository_init + 389
  6   doppllib                            0x000000010b234884 new_SOARepository_init + 52
  7   doppllib                            0x000000010b234300 +[SOARepository initialize] + 64
  8   libobjc.A.dylib                     0x000000010e2bfa14 CALLING_SOME_+initialize_METHOD + 19
  9   libobjc.A.dylib                     0x000000010e2bfdae _class_initialize + 276
  10  libobjc.A.dylib                     0x000000010e2c65e7 lookUpImpOrForward + 225
  11  libobjc.A.dylib                     0x000000010e2d63d4 _objc_msgSend_uncached + 68
  12  doppllib                            0x000000010b234369 SOARepository_get + 57
  13  doppllib                            0x000000010b2338a0 -[SOAQuestionsViewModel register__WithSOAQuestionsViewModel_Host:] + 48
  14  iosApp                              0x000000010aa09edc _T06iosApp14ViewControllerC11viewDidLoadyyF + 220
  15  iosApp                              0x000000010aa0a004 _T06iosApp14ViewControllerC11viewDidLoadyyFTo + 36
  16  UIKit                               0x000000010ecf88a5 -[UIViewController loadViewIfRequired] + 1235
  17  UIKit                               0x000000010ecf8cf2 -[UIViewController view] + 27
  18  UIKit                               0x000000010ebc6b83 -[UIWindow addRootViewControllerViewIfPossible] + 122
  19  UIKit                               0x000000010ebc728b -[UIWindow _setHidden:forced:] + 294
  20  UIKit                               0x000000010ebda208 -[UIWindow makeKeyAndVisible] + 42
  21  UIKit                               0x000000010eb4c8cd -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 4711
  22  UIKit                               0x000000010eb51afd -[UIApplication _runWithMainScene:transitionContext:completion:] + 1720
  23  UIKit                               0x000000010ef172e0 __111-[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:]_block_invoke + 924
  24  UIKit                               0x000000010f2e9be0 +[_UICanvas _enqueuePostSettingUpdateTransactionBlock:] + 153
  25  UIKit                               0x000000010ef16ed9 -[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:] + 249
  26  UIKit                               0x000000010ef17737 -[__UICanvasLifecycleMonitor_Compatability activateEventsOnly:withContext:completion:] + 668
  27  UIKit                               0x000000010f88bc4d __82-[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]_block_invoke + 262
  28  UIKit                               0x000000010f88bb06 -[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:] + 444
  29  UIKit                               0x000000010f56cd52 __125-[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]_block_invoke + 420
  30  UIKit                               0x000000010f769a7e _performActionsWithDelayForTransitionContext + 100
  31  UIKit                               0x000000010f56cb4e -[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:] + 231
  32  UIKit                               0x000000010f2e923d -[_UICanvas scene:didUpdateWithDiff:transitionContext:completion:] + 392
  33  UIKit                               0x000000010eb5037e -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 523
  34  UIKit                               0x000000010f124c1b -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 369
  35  FrontBoardServices                  0x0000000112584158 -[FBSSceneImpl _didCreateWithTransitionContext:completion:] + 338
  36  FrontBoardServices                  0x000000011258cc4d __56-[FBSWorkspace client:handleCreateScene:withCompletion:]_block_invoke_2 + 235
  37  libdispatch.dylib                   0x0000000111d0b43c _dispatch_client_callout + 8
  38  libdispatch.dylib                   0x0000000111d10af4 _dispatch_block_invoke_direct + 592
  39  FrontBoardServices                  0x00000001125b8672 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 24
  40  FrontBoardServices                  0x00000001125b8328 -[FBSSerialQueue _performNext] + 464
  41  FrontBoardServices                  0x00000001125b8897 -[FBSSerialQueue _performNextFromRunLoopSource] + 45
  42  CoreFoundation                      0x000000011160b231 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
  43  CoreFoundation                      0x00000001116aae41 __CFRunLoopDoSource0 + 81
  44  CoreFoundation                      0x00000001115efb49 __CFRunLoopDoSources0 + 185
  45  CoreFoundation                      0x00000001115ef12f __CFRunLoopRun + 1279
  46  CoreFoundation                      0x00000001115ee9b9 CFRunLoopRunSpecific + 409
  47  GraphicsServices                    0x00000001132679c6 GSEventRunModal + 62
  48  UIKit                               0x000000010eb535e8 UIApplicationMain + 159
  49  iosApp                              0x000000010aa0b3e7 main + 55
  50  libdyld.dylib                       0x000000010dbc9d81 start + 1
  51  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type JavaLangIllegalArgumentException
(lldb)
```

However, the `iosTest` tests work. I do not know what `loader` is.

It is entirely possible that this code is not yet supposed to work &mdash; I never tried running it previously. However, it dawned on me that somebody probably *would* try running it, and I expected no crashes (but no visible results, since we're not doing anything "for realz").

**Comment by kpgalligan** [comment]
Need to add 'prefixes.properties' to the main app bundle. This error is because of that:

'JavaLangIllegalArgumentException', reason: 'java.lang.IllegalArgumentException:
interface co.doppl.so.api.SOInterface **is not visible from class loader**'

**Comment by commonsguy** [comment]
Ah, OK, I don't have that in the instructions. I'll fix that as part of the 0.10.5 updates. Thanks!
